nodejs/week3/src/backend/api/meals.js

```diff
@@ -0,0 +1,242 @@
+        console.log(error);    // throw error;
+      }
+    return response.json(results);
+  });
+ 
+}
+
+
+
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++	// Updates the meal by id +	PUT request + api/meals/2
+
+
+const updateMealById = function(request, response){
+  console.log("Updates meal by id with PUT request");
+
+  // const values = request.params.id;
+  const idval = request.params.id;
+  const values = request.body;
+  const {title, description, location, when, max_reservations, price, created_date} = values;
+  
+  // console.log(request.body);
+  
+    // Operation :  select column where id is given
+    pool.query('UPDATE meal SET ? WHERE id = ?', [values, idval], function(error, results, fields) {
+      if (error) {
+        console.log(error);    // throw error;
+      }
+    return response.json(results);
+  });
+ 
+}
+
+
+// +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++	// Deletes the meal by id +	DELETE request + meals/2
+
+
+const deleteMealById = function (request, response){
+  // for looking the request type
+      console.log("Deletes the meal by id with DELETE request");
+
+      const idval = request.params.id;
+      // const values = request.body;
+  
+      // Operation : Delete meals where id is given
+      pool.query('DELETE FROM meal WHERE id = ?', idval, function(error, results, fields) {
+          if (error) {
+            console.log(error);    // throw error;
+          }
+        return response.json(results);
+      });
+  
+  }
+
+
+// +++++++++++++++++++++++++++++++++++++++++	// Get meals that has a price smaller than maxPrice + api/meals?maxPrice=50
+
+
+const MealMaxPrice = function(request, response){
+  console.log("Get meals by maxPrice with GET request");
+
+  const values = request.params.maxPrice;
+  console.log(values);
+  
+    // Operation :  select column where id is given
+    pool.query('SELECT * FROM meal WHERE price < ?', values, function(error, results, fields) {
+      if (error) {
+        console.log(error);    // throw error;
+      }
+    return response.json(results);
+  });
+ 
+}
+
+
+// +++++++++++++++++++++++++++++++++++++++++	// Get meals that still has available reservations + api/meals?availableReservations=true
+
+const MealWithReservation = function(request, response){
+
+  const values = request.query.availableReservations; // true
+  // const currentdate =
+  console.log(values);
+  
+    // Operation :  select column where id is given
+
+    if(values === 'true'){              
+      pool.query('SELECT *, meal.max_reservations, reservation.number_of_guests from meal join reservation on meal.id = reservation.meal_id where max_reservations > number_of_guests', function(error, results, fields) {
+          if (error) {
+            console.log(error);    // throw error;
+          }
+        // return response.send("This works");
+        const leftusers = Object.number_of_guests - Object.max_reservations;
```

Comment:
What is `Object.number_of_guests` and `Object.max_reservations`?

Reply:
it supposed to be total number of guests which are allowed and the maximum numbers which can be reserved so that it can give me how much guests can still reserve the spots. But it's useless.